### PR TITLE
WFCORE-3607 WFCORE-3606 [JDK9 & 10] marshalling upgrade & fixes

### PR DESCRIPTION
**Pull request by ctomc**
WFCORE-3607 [jdk10] upgrade jboss-marshalling to make remoting work on JDK10
WFCORE-3606 [jdk9+] use release=8 for compiling deployment scanner

### CODE REVIEW

**Comment by ctomc** [comment]
this superceeds #3104

File: cli/pom.xml [hunk 1]
```diff
@@ -42,10 +42,6 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
                 <configuration>
-                    <includes>
-                        <include>**/*TestCase.java</include>
-                    </includes>
-                    <enableAssertions>false</enableAssertions>
```
Comment (bstansberry) [inline cli/pom.xml]: This seems unrelated.
